modules/apps/portal/portal-http-impl/src/main/java/com/liferay/portal/http/internal/HttpImpl.java

```diff
@@ -306,7 +316,44 @@ public String URLtoString(URL url) throws IOException {
 	}
 
 	@Activate
-	protected void activate() {
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		HttpConfiguration httpConfiguration =
+			ConfigurableUtil.createConfigurable(
+				HttpConfiguration.class, properties);
+
+		int keepAliveTimeout = httpConfiguration.keepAliveTimeout();
+
+		if (keepAliveTimeout > 0) {
+			_connectionKeepAliveStrategy = new ConnectionKeepAliveStrategy() {
```

Comment:
@ericyanLr Please move the initialization of ConnectionKeepAliveStrategy to _createCloseableHttpClient

Reply:
Hi @tinatian ,

The **ConnectionKeepAliveStrategy** is used in two places: 
**_createCloseableHttpClient** and **_createProxyCloseableHttpClient**.

**Question**
Should I still move the initialization to both those two methods? 

Also, if I move the initialization of the **ConnectionKeepAliveStrategy**, I would still need to maintain the value for **keepAliveTimeout** from the OSGI configuration and create a volatile int variable.

In case this helps, here is also a discussion with YuHai on this in an older PR review:
https://github.com/liferay-core-infra/liferay-portal/pull/6346#discussion_r1433749352